app/src/main/java/com/quran/labs/androidquran/widgets/HighlightingImageView.java

```diff
@@ -294,6 +477,28 @@ protected void onSizeChanged(int w, int h, int oldw, int oldh) {
     }
   }
 
+  private boolean alreadyHighlightedContains(String ayah) {
+    // ayah can either be like 89:7 or 89:7->89:8
+    if(alreadyHighlighted.contains(ayah)) {
+      return true;
+    }
+    int arrowIndex = ayah.indexOf("->");
+    if (arrowIndex == -1) {
+      return false;
+    }
+    String startAyah = ayah.substring(0, arrowIndex);
+    String endAyah = ayah.substring(arrowIndex + 2);
+    if(alreadyHighlighted.contains(startAyah)
+        // TODO: to show or not to show is the question!
```

Comment:
as a rule of thumb, i'd vote for - when in doubt, don't animate - or when it's complicated to figure it out, don't

Reply:
Yes, this method sounds good. I have done exactly that here. 

Say we are playing ayah `a` right now which is already highlighted by the bookmark highlight. Now, when we will switch to playing ayah `b`, then there will be no animation. Only, when both `a` and `b` aren't highlighted, do we show the animation.